tests/robot-tests/tests/libs/utilities.py

```diff
@@ -405,19 +407,9 @@ def user_checks_selected_list_label(list_locator, label):
             f'Selected label "{selected_label}" didn\'t match label "{label}" for list "{list_Locator}"')
 
 
-def user_checks_details_dropdown_contains_publication(details_heading, publication_name):
-    try:
-        sl.driver.find_element_by_xpath(
-            f'//*[contains(@class,"govuk-details__summary-text") and text()="{details_heading}"]')
-    except:
-        raise_assertion_error(f'Cannot find details component "{details_heading}"')
-
-    try:
-        sl.driver.find_element_by_xpath(
-            f'//*[contains(@class,"govuk-details__summary-text") and text()="{details_heading}"]/../..//*[text()="{publication_name}"]')
-    except:
-        raise_assertion_error(
-            f'Cannot find publication "{publication_name}" inside details component "{details_heading}"')
+def user_waits_until_details_dropdown_contains_publication(details_heading, publication_name, timeout=3):
+    sl.wait_until_page_contains_element(f'xpath://*[contains(@class,"govuk-details__summary-text") and text()="{details_heading}"]', timeout=timeout)
```

Comment:
Would say these are too dependent on the class names (bad in general), would change them to include `xpath://details/summary[contains(., "{details_heading}")]`

Reply:
I've made it match the heading exactly just in case there are similar details dropdowns headings on the same page.